packages/repo/src/verify.ts

```diff
@@ -1,52 +1,132 @@
 import { CID } from 'multiformats/cid'
-import * as auth from '@atproto/auth'
-import { RepoStorage } from './storage'
+import { DidResolver } from '@atproto/did-resolver'
+import { ReadableBlockstore, RepoStorage } from './storage'
+import DataDiff from './data-diff'
+import SyncStorage from './storage/sync-storage'
+import ReadableRepo from './readable-repo'
 import Repo from './repo'
-import { DataDiff } from './mst'
-import { def } from './types'
+import CidSet from './cid-set'
+import { parseRecordKey } from './util'
+import { RepoContents } from './types'
+import { def } from '@atproto/common'
 
-export const verifyUpdates = async (
+export type VerifiedCheckout = {
+  contents: RepoContents
+  newCids: CidSet
+}
+
+export const verifyCheckout = async (
+  storage: ReadableBlockstore,
+  root: CID,
+  didResolver: DidResolver,
+): Promise<VerifiedCheckout> => {
+  const repo = await ReadableRepo.load(storage, root)
+  const validSig = await didResolver.verifySignature(
```

Comment:
I will say, I don't love having to thread the `didResolver` through all the sync & verify methods. But it's either that or doing the resolution beforehand & threading through the approved signingKey.
It also doesn't take into account the fact that a user's signing key can change yet. (& god forbid their DID changes, we just error out on that for now)

Reply:
Yeah, I'm not seeing any clearly better options unless there's a natural default `DidResolver`, sort of similar to how in node there's a [global/default `http.Agent`](https://nodejs.org/api/http.html#httpglobalagent).  Can also probably be smoothed over for consumers by assigning a `DidResolver` early on and threading it down here for them in some wrapper/convenience classes, etc.